src/mscorlib/src/System/ThrowHelper.cs

```diff
@@ -120,9 +120,9 @@ internal static void ThrowAddingDuplicateWithKeyArgumentException(object key)
             throw GetAddingDuplicateWithKeyArgumentException(key);
         }
 
-        internal static void ThrowKeyNotFoundException()
+        internal static void ThrowKeyNotFoundException(object key)
         {
-            throw new KeyNotFoundException();
+            throw new KeyNotFoundException(key.ToString());
```

Comment:
@Anipik This needs to use `SR.Format(SR.Arg_KeyNotFoundWithKey, key.ToString())` to get the proper error message.

Also, this may need to split into the `Throw` + `Get...Exception` method pairs like the other similar methods in this file to allow JIT to generate better code for it.

Reply:
isn't both throw  + get already separated as we have ThrowKeyNotFoundException(object key) and KeyNotFoundException(SR.Format(SR.Arg_KeyNotFoundWithKey, key.ToString())); functions

Reply:
Separating it even more has additional benefits - check the discussion in https://github.com/dotnet/coreclr/pull/7580 that introduced it.